pkgs/top-level/python-packages.nix

```diff
@@ -15567,6 +15578,27 @@ in {
     };
   };
 
+  pipenv = buildPythonPackage rec {
```

Comment:
@Mic92 this is not a library, but a cli tool. The header says to keep cli tools in this file

Reply:
It says:

```
Python packages that do not need to be available for each interpreter version do not belong in this packages set.
```

which means not in `python-packages.nix` at all.

For example `gpodder` is a podcast client written in python and we put it not into `python-packages.nix` since we only need one version of it build against one python version.
Looking at the documentation https://github.com/kennethreitz/pipenv#basic-concepts it seems we also only need one version of `pipenv` for both python2/python3. In that case please add it to `top-level/all-packages.nix`

Reply:
Uhm, I think that the text in the header should be updated then, since it says

> Expressions for Python libraries are supposed to be in `pkgs/development/python-modules/<name>/default.nix`.

If libraries aren't supposed to be in `python-packages.nix` at all, that leaves only cli application...

I presume that if neither cli applications nor libraries are supposed to live in `python-packages.nix`, then this file still exists only for legacy reasons?

Reply:
We still add python libraries there, but are using `callPackage`, where the expression is written to
 python-modules, i.e:  `email_validator = callPackage ../development/python-modules/email-validator { };`